pkg/sentry/kernel/semaphore/semaphore.go

```diff
@@ -230,77 +207,55 @@ func (r *Registry) HighestIndex() int32 {
 	return highestIndex
 }
 
-// RemoveID removes set with give 'id' from the registry and marks the set as
+// Remove removes set with give 'id' from the registry and marks the set as
 // dead. All waiters will be awakened and fail.
-func (r *Registry) RemoveID(id int32, creds *auth.Credentials) error {
+func (r *Registry) Remove(id ipc.ID, creds *auth.Credentials) error {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	set := r.semaphores[id]
-	if set == nil {
-		return syserror.EINVAL
-	}
+	r.reg.Remove(id, creds)
+
 	index, found := r.findIndexByID(id)
 	if !found {
 		// Inconsistent state.
 		panic(fmt.Sprintf("unable to find an index for ID: %d", id))
 	}
-
-	set.mu.Lock()
-	defer set.mu.Unlock()
-
-	// "The effective user ID of the calling process must match the creator or
-	// owner of the semaphore set, or the caller must be privileged."
-	if !set.checkCredentials(creds) && !set.checkCapability(creds) {
-		return syserror.EACCES
-	}
-
-	delete(r.semaphores, set.ID)
 	delete(r.indexes, index)
-	set.destroy()
+
 	return nil
 }
 
-func (r *Registry) newSet(ctx context.Context, key int32, owner, creator fs.FileOwner, perms fs.FilePermissions, nsems int32) (*Set, error) {
+func (r *Registry) newSet(ctx context.Context, key ipc.Key, creator fs.FileOwner, perms fs.FilePermissions, nsems int32) (*Set, error) {
```

Comment:
Rename function to `newSetLocked` to indicate `r.mu` must be held by the caller, and document the requirement as a precondition.

Reply:
Done. Also updated `newShm` to match `set` and `msgqueue`.